### PR TITLE
ValueMappings: Fixes issue with regex value mapping that only sets color

### PR DESCRIPTION
**Pull request by torkelo**
When investigating an issue I noticed that regex value mappings always matched and replaced the matched element even when
the replace with text was empty (which it is when you only want to set color).

### CODE REVIEW

**Review by ashharrison90 — COMMENTED** [review]
tiny question

File: packages/grafana-data/src/utils/valueMappings.ts [hunk 1]
```diff
@@ -58,9 +58,13 @@ export function getValueMappingResult(valueMappings: ValueMapping[], value: any)
 
         const regex = stringToJsRegex(vm.options.pattern);
         if (value.match(regex)) {
-          const thisResult = Object.create(vm.options.result);
-          thisResult.text = value.replace(regex, vm.options.result.text || '');
-          return thisResult;
+          const res = { ...vm.options.result };
+
+          if (res.text != null) {
```
Comment (ashharrison90) [inline packages/grafana-data/src/utils/valueMappings.ts:63]: i think `!=` will also mean we don't go into this branch if `res.text === undefined`. is that what we want, or should this be `!==`? 🤔

Comment (torkelo) [reply packages/grafana-data/src/utils/valueMappings.ts:63]: yes, the != is deliberate to also exclude undefined

**Review by ashharrison90 — APPROVED** [review]
lgtm then 🙌

**Pull request merged**